### ISSUE TITLE
[RFC] How about required one or more relations?

### ISSUE THREAD

**Issue by Largo** [bolt/bolt#3734]
I'm finding myself in need of a validation so that a user cannot save a record without adding at least one relation.

Real life example: I'm creating a website for a small business. The boss is an M$ Office trainer, so she's offering office training courses. They are linked to a category record (Excel, Word, Powerpoint ...)
Now, it's easy to forget to add the relation, while entering. If you have many courses, they might not show up. 

I'm proposing to add the "required: true"  to the relationship definition to make the field mandatory.
**TL;DR: Feature to show an error, if a relation is empty**

(btw. Regarding the RFC: I've sadly don't have time to work on this anytime soon. Moving to Japan from Europe next month ^^'')

**Comment by GawainLynch** [comment]
> Regarding the RFC: I've sadly don't have time to work on this anytime soon. Moving to Japan from Europe next month

Enjoy! Japan is a remarkable place.

**Comment by rarila** [comment]
Konnichiwa! :jp:

**Comment by rlowry** [comment]
I recently added this for Relations and Taxonomy -> Categories. This works for my requirements but it's probably not the "proper" way it should be implemented...

A small edit to the following files is required (last line in each block);

**Relations**
\vendor\bolt\bolt\app\view\twig\editcontent\fields\_relationship.twig
```twig
{# BUIC options #}
{% set buic_opt_select = {
    'name':     'relation[' ~ relcontenttype ~ '][]',
    'id':       'relation-' ~ relcontenttype,
    'multiple': relation.multiple is defined and relation.multiple == 1,
    'options':  options,
    'clear':    true,
    'required':  relation.required|default(false),
} %}
```

**Taxonomy -> Categories**
\vendor\bolt\bolt\app\view\twig\editcontent\fields\_categories.twig

```twig
{# BUIC options #}
{% set buic_opt_select = {
    'all':       taxonomy.multiple|default(false),
    'clear':     true,
    'id':        'taxonomy-' ~ taxonomy.slug,
    'multiple':  taxonomy.multiple|default(false),
    'name':      'taxonomy[' ~ taxonomy.slug ~ '][]',
    'options':   options,
    'required':  taxonomy.required|default(false),
} %}
```
You can then add the required option to your relations and taxonomy -> categories when defining them in contenttypes.yml and taxonomy.yml respectively. e.g.

**contenttypes.yml**
```yml
products:
    name: Products
    singular_name: Product
    fields:
        name:
            type: text
            group: details
            required: true
        slug:
            type: slug
            uses: name
            required: true
    relations:
        collections:
            multiple: false
            label: Select a collection
            required: true
    taxonomy: [ categories ]
    default_status: published
```
**taxonomy.yml**
```yml
categories:
    name: Categories
    slug: categories
    singular_name: Category
    singular_slug: category
    behaves_like: categories
    options: [ Socks, Shoes, Shirts]
    required: true
```

**Comment by SahAssar** [comment]
@rlowry would you be willing to PR that and get a sponsor to work to get it into core?

**Comment by rlowry** [comment]
@SahAssar sorry for late reply. I'm not sure when I'd be able to work on this further so wouldn't want to commit at this time. Happy for anyone else to take this forward. My hack is a quick and dirty solution and would not be suitable for a number of use cases.

**Comment by bobdenotter** [comment]
Because of "Spring cleaning", we're going to close this issue. Unless somebody steps up to help out with implementing it, it's not expected that the core team will have the opportunity to work on this soon. Keeping it open might set false expectations for people stumbling upon this issue.

This is not because we don't see this feature as potentially worthwhile, it is more to do with an overload of potential new features that could be developed combined with non-limitless amounts of available time and resources. 

If you disagree, feel free to re-open for discussion.
